command/harness/setup.go

```diff
@@ -222,5 +230,11 @@ func HandleSetup(ctx context.Context, r *SetupVMRequest, s store.StageOwnerStore
 
 	logr.WithField("response", fmt.Sprintf("%+v", setupResponse)).Traceln("VM setup is complete")
 
+	deleteInfoMap["shouldDelete"] = "true"
+	if env.Runner.CleanupDebug {
+		logr.Traceln("shouldDelete set to true")
+		util.CreateEnvState().Add(stageRuntimeID, deleteInfoMap)
```

Comment:
envState is used for step output envs. We should not use the same for this purpose.

Reply:
I don't see any issue if we reuse this here. Besided it's a just a custom tag for customer test

Reply:
every env output for a step will have these values which should not be present.

Reply:
got it, Will create a new class for it.